File: JavaScript_problems/group_anagrams.js
var groupAnagrams = function(strs) {
    const obj = {}
    
    strs.forEach(str => {
        const sortedString = str.split("").sort().join("")
        
        if (!obj[sortedString]){
            obj[sortedString] = [str]
        } else {
            obj[sortedString].push(str)
        }
    })
    
    const arr = []
    for (const key in obj){
        arr.push(obj[key])
    }
    
    return arr
};

File: ruby_problems/group_anagrams.rb
def group_anagrams(strs)
    
    hash = {}
    
    strs.each do |str|
        sortedStr = str.chars.sort.join
        
        if (!hash[sortedStr])
            hash[sortedStr] = [str]
        else
            hash[sortedStr] << str
        end
    end
    
    arr = []
    hash.each_value do |val|
        arr << val
    end
    
    arr
end